nessie/src/main/java/org/apache/iceberg/nessie/UpdateableReference.java

```diff
@@ -60,6 +60,8 @@ public String getHash() {
     return reference.getHash();
   }
 
+  /** @deprecated will be removed in 1.3.0 */
+  @Deprecated
   public Branch getAsBranch() {
```

Comment:
Don't understand why the existing call sites are changed.

Reply:
because all the places where this was called, already there is a check for `getRef().checkMutable()`.  So, no need for one more method calls that check if it is a branch. Hence, directly casting to branch.